lib/synapse_pay_rest/models/transaction/transaction.rb

```diff
@@ -213,7 +213,7 @@ def add_comment(comment)
         trans_id: id,
         payload: payload
       )
-      self.class.from_response(node, response)
```

Comment:
This is intended. v3.0.0 of the library is meant to coincide with a minor API update which will make the PATCH /trans response more consistent with other responses, by returning a transaction object.